unit_tests/MacroTranslTests.d

```diff
@@ -0,0 +1,457 @@
+/**
+ * Copyright: Copyright (c) 2016 Wojciech Szęszoł. All rights reserved.
+ * Authors: Wojciech Szęszoł
+ * Version: Initial created: Jun 02, 2016
+ * License: $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost Software License 1.0)
+ */
+
+import Common;
+
+import dstep.translator.Context;
+import dstep.translator.MacroDefinition;
+import dstep.translator.Options;
+import dstep.translator.Output;
+
+void assertTranslatesMacroDefinition(
+    string expected,
+    string source,
+    string file = __FILE__,
+    size_t line = __LINE__)
+{
+    auto translUnit = makeTranslationUnit(source);
+
+    Options options;
+    Output output = new Output;
+
+    Context context = new Context(translUnit, options);
+
+    context.macroLinkage = "";
+
+    auto children = translUnit.cursor.children(true);
+
+    assert(children.length == 1);
+
+    translateMacroDefinition(output, context, children[0]);
+
+    assertEq(expected, output.data, false, file, line);
+}
+
+private alias assertTMD = assertTranslatesMacroDefinition;
+
+// Translate basic macro definitions.
+unittest
+{
+    assertTMD(q"D
+D", q"C
+#define FOO
+C");
```

Comment:
Isn't it more natural to have the C code as the first parameter? Like here https://github.com/jacob-carlborg/dstep/blob/master/unit_tests/UnitTests.d#L14-L16.